### PR TITLE
Add double-star mutli-level directory glob matching to filelog receiver

### PR DESCRIPTION
**Pull request by pmalek-sumo**
Closes: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2329

### CODE REVIEW

File: operator/builtin/input/file/config_test.go [hunk 1]
```diff
@@ -80,6 +80,24 @@ func TestConfig(t *testing.T) {
...
+				cfg.Include = append(cfg.Include, "**.log")
+				return cfg
+			}(),
+		},
+		{
+			"include_glob_double_asterisk_nested",
+			false,
+			func() *InputConfig {
+				cfg := defaultCfg()
+				cfg.Include = append(cfg.Include, "directory/**/*.log")
```
Comment (djaglowski) [inline operator/builtin/input/file/config_test.go:97]: Could we add one more test for `**/directory/**/*.log`?

**Review by djaglowski — COMMENTED** [review]
Looks good, just a couple minor suggestions.

Could you also update the `file_input` documentation so that users can easily find the exact rules that govern paths. It's ok to directly reference the [doublestar docs](https://github.com/bmatcuk/doublestar#patterns). This is done elsewhere, such as in the [expressions](https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/types/expression.md) doc.

File: operator/builtin/input/file/config_test.go (hunk 1 above)
Comment (pmalek-sumo) [reply operator/builtin/input/file/config_test.go:97]: Done

**Review by djaglowski — APPROVED** [review]
Thanks @pmalek-sumo. LGTM

**Pull request merged**